autoPyTorch/datasets/base_dataset.py

```diff
@@ -57,19 +117,27 @@ def _get_indices(self) -> np.ndarray:
         return indices
 
     def create_cross_val_splits(self,
-                                cross_val_type: str,
-                                num_splits: int) -> List[Tuple[Dataset, Dataset]]:
-        if cross_val_type not in self.cross_validators:
-            raise NotImplementedError(f'The selected `cross_val_type` "{cross_val_type}" is not implemented.')
-        kwargs = {}
-        if is_stratified(cross_val_type):
-            # we need additional information about the data for stratification
-            kwargs["stratify"] = self.train_tensors[-1]
-        splits = self.cross_validators[cross_val_type](num_splits, self._get_indices(), **kwargs)
-        return [(Subset(self, split[0]), Subset(self, split[1])) for split in splits]
+                                cross_val_type: CrossValTypes,
+                                num_splits: int) -> None:
+        """
+        This function creates the cross validation split for the given task.
+
+        It is done once per dataset to have comparable results among pipelines
+        """
+        # Create just the split once
+        if self.splits is None:
```

Comment:
Maybe you could add an `else:` with a warning/error when this is called after already creating a split?

Reply:
I have clarified why I do expect this function to be called multiple times via a comment:
```
      # Create just the split once
         # This is gonna be called multiple times, because the current dataset
         # is being used for multiple pipelines. That is, to be efficient with memory
         # we dump the dataset to memory and read it on a need basis. So this function
         # should be robust against multiple calls, and it does so by remembering the splits

```